firebase.go

```diff
@@ -128,8 +145,43 @@ func NewApp(ctx context.Context, config *Config, opts ...option.ClientOption) (*
 
 	return &App{
 		creds:         creds,
+		databaseURL:   config.DatabaseURL,
 		projectID:     pid,
 		storageBucket: config.StorageBucket,
 		opts:          o,
 	}, nil
 }
+
+func amendDefaultConfig(config *Config) (*Config, error) {
```

Comment:
I'm not thrilled with the name "amend" here, but I don't have a great alternative. Please add a comment explaining that this fills in values missing from the passed in config (by mutating it), if they are present in the JSON file specified by the env variable.

I'm not sure it's helpful for the function to return a config, since it's always the config that was passed in. I suspect that's more likely to just be confusing. Maybe it should just return `error`?